src/main/java/org/commcare/formplayer/application/FormSessionFactory.java

```diff
@@ -39,25 +44,28 @@ public class FormSessionFactory {
     @Autowired
     private CommCareSessionFactory commCareSessionFactory;
 
-    public FormSession getFormSession(SerializableFormSession serializableFormSession, String windowWidth) throws Exception {
+    public FormSession getFormSession(SerializableFormSession serializableFormSession, String windowWidth, boolean keepAPMTraces) throws Exception {
         CommCareSession commCareSession = commCareSessionFactory.getCommCareSession(serializableFormSession.getMenuSessionId());
-        return getFormSession(serializableFormSession, commCareSession, windowWidth);
+        return getFormSession(serializableFormSession, commCareSession, windowWidth, keepAPMTraces);
     }
 
     @NotNull
     public FormSession getFormSession(SerializableFormSession serializableFormSession,
-            @Nullable CommCareSession commCareSession, @Nullable String windowWidth) throws Exception {
+            @Nullable CommCareSession commCareSession, @Nullable String windowWidth, @Nullable boolean keepAPMTraces) throws Exception {
         FormplayerRemoteInstanceFetcher formplayerRemoteInstanceFetcher = new FormplayerRemoteInstanceFetcher(
                 runnerService.getCaseSearchHelper(),
                 virtualDataInstanceService);
+        HashMap<String, Object> metaSessionContext = new HashMap<String, Object>();
+        metaSessionContext.put(WINDOW_WIDTH, windowWidth);
+        metaSessionContext.put(KEEP_APM_TRACES, keepAPMTraces);
```

Comment:
probably a better design to construct the `sessionContext` from the controller itself.